src/box/iproto.cc

```diff
@@ -985,8 +1019,6 @@ iproto_connection_close(struct iproto_connection *con)
 		con->state = IPROTO_CONNECTION_CLOSED;
 	} else if (con->state == IPROTO_CONNECTION_PENDING_DESTROY) {
 		iproto_connection_try_to_start_destroy(con);
-	} else {
-		assert(con->state == IPROTO_CONNECTION_CLOSED);
```

Comment:
Why?

Reply:
On connection drop we call `iproto_connection_close` for connection in any state. Let's call only for alive.

```diff
@@ -3491,7 +3493,8 @@ iproto_do_cfg_f(struct cbus_call_msg *m)
                                                {{ tx_process_cancel, NULL }};
                rlist_foreach_entry(con, &iproto_thread->connections,
                                    in_connections) {
-                       if (!con->is_in_replication)
+                       if (!con->is_in_replication &&
+                           con->state == IPROTO_CONNECTION_ALIVE)
                                iproto_connection_close(con);
                        con->is_drop_pending = true;
                        iproto_thread->drop_pending_connection_count++;
```